deps/rabbit/src/tcp_listener.erl

```diff
@@ -63,12 +63,20 @@ start_link(IPAddress, Port,
 
 %%--------------------------------------------------------------------
 
-init({IPAddress, Port, {M,F,A} = OnStartup, OnShutdown, Label}) ->
+init({IPAddress, Port, {M, F, A} = OnStartup, OnShutdown, Label}) ->
+    ensure_credential_obfuscation_secret(),
+
     process_flag(trap_exit, true),
     logger:info("started ~s on ~s:~p", [Label, rabbit_misc:ntoab(IPAddress), Port]),
     apply(M, F, A ++ [IPAddress, Port]),
-    {ok, #state{on_startup = OnStartup, on_shutdown = OnShutdown,
-                label = Label, ip=IPAddress, port=Port}}.
+    State0 = #state{
+        on_startup = OnStartup,
```

Comment:
Do we even need to store `on_startup`? It's not used. I would just remove it from the record.